### DISCUSSION TITLE
Keine Einspeisung durch Batterie bei Netzbezug !?

### DISCUSSION THREAD

**Discussion by plauzi45** [hoylabs/OpenDTU-OnBattery#323] [General]
Hallo zusammen,
ich bin neu hier bei Github, erst mal vielen Dank für die OpenDTUon Battery erweiterung an die Macher !

Leider bekomme ich die DTU nicht so eigenstellt das Sie bei mir funktioniert.

Mein Aufbau ist folgender. Ich lade Tagsüber einen Lipofe4 24V und 100ah direkt über einen Victron Laderegler.
Der Hoymiles WR hängt dann direkt an der Battery und wird Nachts über eine smarte Steckdose zugeschaltet.
Bisher habe ich die Einspeisungleistung über ein Node Red Script Zeitgesteuert geregelt.

Ich würde die Einspeisung aber gerne Lastabhängig machen.

Daher habe ich die OpenDTUon Battery geflasht.

Den Verbrauch übergebe ich aus Home Assistant mit MQTT über ein Node Red Script. Ursprünglich kommen die Werte aus einem MODBUS TC Energiemeter.

Jetzt aber zu meinem Problem. Egal welche Werte ich im Dynamic Power Limiter einstelle, der WR übernimmt immer nur die Positiven Netzeinspeisewerte als Limit. ( Die sind bei mir Positiv da ich noch eine PV besitze ) Die Negativen Werte quasi wenn ich Netzbezug habe werden nicht als Limit weitergegeben.
Hinzu kommt noch das die Positiven Werte in voller Höhe weitergegeben werden als Limit. Dadurch entsteht auch schon mal ein Limit von 1024W.

Hat einer eine Idee was ich falsch mache ?

Vielen Dank im voraus !

Gruß Dennis

**Comment by spcqike** [comment]
Ich bin mir nicht ganz sicher aber wenn ich dich richtig verstehe ist dein Wert „-512“ wenn du 512W aus dem Netz beziehst, und „256“ wenn du 256W einspeist. 

Wenn ja, wird das dein Problem sein. 

Die Firmware erwartet positive Werte für Bezug und negative für eine Speisung. 

Versuch einfach mal dein power wert im Node red zu negieren bevor du es per MQTT verteilst.

**Comment by plauzi45** [comment]
Danke für den Tipp ! ich habe das jetzt mal mit einer Node RED funktion von Negativ in Positiv umgerechnet, das funktioniert auch
dann.
Allerdings nimmt die DTU die Limits nicht an.
Es steht dann bei 1100 W Bezug auch 1100 Watt Limit !?

**Reply by spcqike** [reply]
Na, was für Limits willst du denn Haben? 

Die DTU versucht eine Nulleinspeisung. Heißt, wenn du aktuell 1100W aus dem Netz ziehst und dein wechselrichter 1100W liefern kann, wird das auch als Limit eingestellt.

**Comment by Phantomias2006** [comment]
@spcqike 
du kannst die Werte in NodeRed noch prüfen und auf Grenzwerte setzen wenn sie über oder unterschreiten.
LG


```
var min = 10;
var max = 50;

msg.payload = clamp(msg.payload, min, max);
return msg;

function clamp(value, min, max) {
	return Math.max(value, min), max);
}
```

**Comment by plauzi45** [comment]
Danke für den Tipp ! So hab ichs jetzt gemacht.
Ich rechne in einer Funktion negtiv in positiv um und
setze dann die Limits.
Ich hatte mir von OPENDTU on Battery noch eine zusätzliche Überwachung
der Batterie erhofft und eine zusätzliche Hysterse, damit die Werte nicht so springen.
Da es aber aus welchen Gründen auch immer nicht funktioniert bleibe ich jetzt beim NODE RED Script

**Reply by Phantomias2006** [reply]
Du kannst natürlich auch in NodeRed eine "Hysterese" machen...
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/356e53a9-5fbd-4c64-9d28-64c3f3a7378c)
Mach einen filter und setze ihn auf
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/15437240/409a1a37-97bf-4f1c-b8e2-e3b5f38a5487)

**Reply by spcqike** [reply]
@plauzi45 poste doch mal deine Einstellung (Screenshot) und beschreibe nochmal dein Ziel. 

Eigentlich kann openDTU on battery eine hysterese und auch die Batterie überwachen. Btw Ladezustände und Spannungen berücksichtigen.

**Comment by plauzi45** [comment]
Danke für eure Hilfe.

Ich habe noch ein bißchen rumprobiert aber das Ergebnis ist immer das gleiche.
Wenn ich das max. und min. Limit in Node Red setze dann werden diese Limits auch
als Limits in die DTU übernommen z.B. max. 600W und min 12W da der WR unter 10W abschaltet.
Liegt der Wert dazwischen wird dieser übergeben und die Leistung pendelt hin und her.

Setze ich die Limits in OPENDTUon Battery dann ist der max. Wert so hoch wie der Bezug z.B. wie im Bild 
2501 und 715%.
Wenn der min., Wert erreicht wird schaltet die DTU den WR ab. Siehe Bild "Shutdown by Remote Kontroll"
Das Problem ist wenn der Wert dann wieder hoch geht bleibt der WR aus. ( Kästchen Gelb hinterlegt. )

![unteres_Limit](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/98e3b213-c3ff-486b-9706-35bd3d320f72)

![2023-07-21_135130](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/e1743a23-c39b-464f-998b-fc4182f0e23e)

![2023-07-21_140143](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/7dfb6238-db4b-42b2-8566-f4ab32692c9d)

![2023-07-21_140155](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/70bf1fdd-702a-4f04-88ae-f10e679b8cc9)

![2023-07-21_140221](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/728ad53f-7cd5-4cc5-ac29-1e7d619a768c)

**Comment by helgeerbe** [comment]
Deine Einstellungen machen auch keinen Sinn. Du hast eine 24 V Battery. Ich würde mal folgende Einstellungen machen:
- DC-Spannung start: 26,4 (Erst bei Erreichen dieser Spannung wird der Wechselrichter wieder eingeschaltet)
- DC-Spannung Stop: 21,6 (20 finde ich schon sehr niedrig)

D.h. fällt die Batteriespannung unter 21,6 V wird der Wechselrichter ausgeschaltet. Wird die Batterie geladen, so wird der Wechselrichter bei 26,4 V wieder eingeschaltet. Du hast da 30V eingestellt, d. h. der Wechselrichter wird nie eingeschaltet.

Die Leistung wird dynamich geregelt, aber nie mehr als 598W,  unter 40W wird der Wechselrichter abgeschaltet (könntest du eigentlich auf 20W setzen). 598 W wirst du real nicht erreichen, da der WR maximal um die 11A pro Eingang zieht (genaue Zahl habe ich jetzt nicht im Kopf) => 11A * 2Eingänge * 24V = 528 W.  Um die vollen 600 W zu bekommen, bräuchtest du ein 48 V System.

Angestrebter Netzbezug ist -20. D. h. es ist für dich OK, immer 20W ins öffentliche Netz einzuspeisen. Ich würde eher 40 W nehmen. D. h. du hast immer leichten Netzbezug. Die Batterie wird über die Zeit sowieso leer. Aber so verhinderst du eine ungewollte Einspeisung.

Der Dynamic Power Limiter sollte dann dynamisch anhand des aktuellen Netzbezug das Limit des Wechselrichters setzen. Falls weniger als 20 W benötigt werden, wird auch abgeschaltet. Das ist das minimum was der Wechselrichter steuern kann.

**Reply by spcqike** [reply]
Kann das Limit den. >100% sein? In seinem letzten Screenshot sieht man, dass er auf dem HM350 ein Limit von 2.501W bzw 715% gesetzt hat. 

Sollte der DPL das am Ende nicht auf 350W begrenzen?

**Reply by helgeerbe** [reply]
Stimmt habe übersehen, dass es ein HM350 ist. Der DPL begrenzt aber auf max Limit des Wechselrichters, in dem Fall sind es dann 350 W.

Die Anzeige ist komisch.  Kann nur sein, wenn das die eingetragenen Werte sind und der DPL noch nie versucht hat etwas zu regeln. Z. B. das die Start-Spannung der Batterie noch nicht erreicht ist. Was bei den obigen Werten von 30 V nie der Fall sein dürfte.

**Comment by plauzi45** [comment]
Danke für die schnellen Antworten.

Die max. und min. Spannung der Batterie habe ich nur zum testen so weit ausseinander gesetzt um auszuschließen das die Abschaltung wegen der Batterie erfolgt. Dabei habe ich mich leider vertan 30 sind natürlich zu hoch. 
Das werde ich direkt mal testen dadurch erklärt sich vielleicht das Abschalten. Aber warum das obere Limit nicht begrenzt wird ist dadurch nicht erklärt.
Das Aktuell ein HM 350 dran hängt ist nur eine Übergangslösung. Es soll Mittelfristig ein 800er oder 1500er dran. Deshalb sind mir auch die Limits so wichtig. 
Da geht es mir Primär drum.
Ich speiße auch an diesem Wechselrichter ausschließlich über die Batterie ein. Tagsüber werde ich von einer großen PV versorgt.
Es geht mir nur um die Nachteinspeisung.

**Reply by spcqike** [reply]
Wie @helgeerbe geschrieben hat. Normalerweise brauchst du das Limit gar nicht manuell setzen. Das macht der DPL (dynamische power limiter) 

Schick der opendtu deinen Stromverbrauch (power Meter per mqtt) und lass die opendtu den Rest regeln. Sie versucht automatisch auf dein power Target zu kommen.

**Comment by plauzi45** [comment]
Habe es getestet mit einer geänderten Startspannung, es passiert aber wieder genau das gleiche.
Sobald der eingestellte min. Wert erreicht wird schaltet die DTU den WR ab.

![minXX](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/432ef524-eda0-41b4-b99d-018fdc37595c)

**Reply by spcqike** [reply]
Du hast ja auch eine Einspeisung von -2.000W. Warum sollte er also zusätzlich einspeisen?

**Reply by plauzi45** [reply]
Soll er ja auch nicht aber die DTU soll den WR auch nicht abschalten.

Zu erkennen an dem gelb hinterlegten kästchen, waäre der WR noch an wäre es blau wie im Anhang

Lege ich in Node RED eine Funktion fest mit min. Wert für den WR bleibt der WR an bei 12W und eingeschaltet , lege ich im Limiter einen
min. Wert fest geht der WR aus und danach nicht mehr an wenn der Bezug von + in - wechselt.


![min_xxxx](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/84d66aa1-6c24-4b0b-9fb2-9a192efc26d8)

**Reply by helgeerbe** [reply]
Du hattest aber bei dir eingetragen, dass er unter 40 W Abschalten soll.

**Reply by helgeerbe** [reply]
Und das abschalten ist richtig, da du den WR nicht auf null setzen kannst. Es gibt einen minimalen Wert, den du nicht unterschreiten kannst. Danach hilft nur abschalten, um nicht kostbare Batterie Energie ins Netz zu pusten

**Reply by schlimmchen** [reply]
@plauzi45 Warum ist es ein Problem für dich, dass der Wechselrichter abgeschaltet wird? Das passiert immer, wenn das errechnete Power Limit unter dem eingestellten "Unteres Leistungslimit:" liegt. Oder anders: wenn der WR weniger als "Unteres Leistungslimit:" erzeugen soll, dann wird er abgeschaltet. Er wird natürlich wieder eingeschaltet, wenn er erzeugen soll.

**Reply by plauzi45** [reply]
Danke noch mal für eure Antworten, ich probiere es noch mal durch, das Problem beim abschalten bei mir ist das er danach nicht wieder angeht wenn sich die Werte ändern. Könnte aber mit den einstellwerten min/max der Batteriespannung zu tun gehabt haben. Das teste ich noch mal. 
Ich hatte den WR über das Node red Script immer so bei min.12W laufen lassen damit er nicht ganz abschaltet.
Dann reagiert er schneller wenn sich der Bezug wieder erhöht.
Wenn er ganz aus ist dauert es nämlich lang bis er wieder hochfährt. Ist das bei euch nicht so ?

**Comment by plauzi45** [comment]
Hi nochmal,

ich habe es jetzt noch mal ausgiebig getestet indem ich Werte aus Node Red an die DTU gesendet habe.
Hat die DTU den WR einmal abgeschaltet weil unter 40W und es wechselt wieder zu Bezug schaltet die DTU
den WR nicht mehr an.
Starte ich den WR manuell läuft es erst mal wieder bis der Wert unter 40W ( min. Limit ) springt. Dann geht der WR nicht mehr an
bis zum nächsten manuellen start durch mich
Keine Ahnung warum

**Reply by spcqike** [reply]
Welche Einstellungen hast du denn aktuell in der openDTU eingetragen? Welche Spannungen und Leistungswerte?

**Reply by plauzi45** [reply]
Siehe Bilder, ich versuche jetzt mal mit Node Red eine klein wenig höhere Untergrenze zu übergeben um das Abschalten zu vermeiden.
Wenn das funktioniert kann ich die DTU austricksen und ggf. wenigstens die Spannungsabschaltung der Batterie nutzen. 

![2023-07-22_122845](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/2ad61159-2a8d-48b5-9f62-ce79b882db9e)
![2023-07-22_122857](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/f88923f8-c99c-4e16-85ac-c31552aa5ff9)
![2023-07-22_122907](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/139256310/4beabe01-cc33-485a-9756-f24c0c99fbbb)

**Comment by spcqike** [comment]
Welche Spannung hat deine Batterie denn, nachdem deine Leistung < 40W war sodass der wechselrichter aus gemacht wurde?

**Reply by plauzi45** [reply]
Die Spannung lag bei ca. 26,6 und fällt beim einschalten kurzzeitig um ca. 0,4V ab bevor die DTU den WR ganz abschaltet.

**Reply by spcqike** [reply]
Kann ich dann grade nicht nachvollziehen. Evtl hilft ein Auszug aus der Konsole in der steht warum er was macht. 

Theoretisch sollte der WR bei einer Spannung von 26.6V und einem netzbezug von 40.1W (vielleicht auch 41W) gestartet werden und versuchen deinen netzbezug auf 0W zu bringen. 

Da du eine hysterese von 50W hast würde er dann aber erst wieder bei 91W eine Änderung machen. Denke ich.

**Reply by plauzi45** [reply]
Nochmal danke für die Hilfe, ich kapituliere und lasse erst mal alles wieder mit Node Red Script laufen.
Ich versuche es noch mal wenn ich einen anderen WR habe oder ein  paar Releases weiter.
